lib/ntpUtil.js

```diff
@@ -126,7 +126,7 @@ const prepareAssets = (jsonFileUrl, targetResourceDir, targetJsonFileName) => {
         return reject(error)
       }
 
-      createPhotoJsonFile(path.join(targetResourceDir, 'photo.json'), JSON.stringify(photoData))
+      createPhotoJsonFile(path.join(targetResourceDir, targetJsonFileName), JSON.stringify(photoData))
```

Comment:
This seems like it could be an issue on first glance without knowing how this is called everywhere. As a quick and simple fix we could validate that the `targetJsonFileNames` and `targetResourceDir` are values we expect within this function to be certain we're not introducing a path traversal vulnerability. 

The issue I'm thinking might be possible (but will require further digging in how this code works) is that a sponsored image is named in a particular way such that it could load an unexpected image (and count it as viewing a sponsored image) or potentially overwrite a file on disk. I'm not certain this is possible at the moment, but have you considered this and handled it else where?